src/validation/validators/fileType/FileTypeValidation.ts

```diff
@@ -0,0 +1,25 @@
+import { InvalidFileTypeError } from '@/validation/errors'
+import { IFieldValidation } from '@/validation/protocols'
+
+export class FileTypeValidation implements IFieldValidation {
+  constructor(
+    readonly field: string,
+    private readonly allowedFileExtensions: string[]
+  ) {}
+
+  private getFileExtension(fileName: string): string {
+    const parts = fileName?.split('.');
+    if (parts?.length === 1 || (parts?.[0] === '' && parts?.length === 2)) {
+      return ''; // No extension found
+    }
+    return parts?.pop()!.toLowerCase();
+  }
+
```

Comment:
instead of having a map loop to convert ext to lowercase and a includes to verify, couldn't you use a find?
like this:

```
validate(input: object): Error {
  const fileExtension = this.getFileExtension((input[this.field] || '').toLowerCase());
  const isAllowed = this.allowedFileExtensions.find((ext) => ext.toLowerCase() === fileExtension);
  return !fileExtension || isAllowed
    ? null
    : new InvalidFileTypeError();
}

```

PS: before PR, run `npm run format:fix` to run prettier

Reply:
Should the return statement be like the below for the specified changes:
```ts
return fileExtension && isAllowed
  ? null
  : new InvalidFileTypeError();
```
because if extension is not provided the `fileExtension` becomes `''` and `isAllowed` also becomes `''` which then evaluates as `True` and results in `null`

Reply:
Hmmm, got it @Arnav-2004